File: Backend-SpringBoot/backend/src/main/java/com/smartlearninghub/dto/certificate/CertificateGenerationResponse.java
package com.smartlearninghub.dto.certificate;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Response payload returned by the ASP.NET Core Certificate microservice
 * after generating a PDF. Field names match the .NET service's JSON contract.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class CertificateGenerationResponse {

    private String certificateNumber;
    private String pdfPath;
    private String downloadUrl;
}

File: CertificateService_DotNet/certificate-service/src/SmartLearningHub.CertificateService.Application/DTOs/CertificateGenerationResponse.cs
namespace SmartLearningHub.CertificateService.Application.DTOs;

/// <summary>
/// Response returned to the Spring Boot backend after a certificate is
/// generated, matching CertificateGenerationResponse.java field names.
/// </summary>
public class CertificateGenerationResponse
{
    public string CertificateNumber { get; set; } = string.Empty;

    public string PdfPath { get; set; } = string.Empty;

    public string DownloadUrl { get; set; } = string.Empty;
}
